### ISSUE TITLE
Unwanted event generator behavior after fixing beamTarget

### ISSUE THREAD

**Issue by cameronc137** [JeffersonLab/remoll#98]
@wdconinc What was the original reason for making this change?

I'm not sure if this is relevant, but we are having problems with the new remollGenRemoll.cc generator producing unwanted events that appear to be from another source somehow. 

Also, when the fVertexPos.SetZ(number) command is not executed in the generator it allows the remollVEventGen.cc to sample a uniform distribution (taken from the aux value declared target volume - I think), but it would be nice if this functionality was able to be turned off in a more elegant way, especially since I suspect this may also be the origin of the strange background of signals we saw in the remollGenRemoll.cc output. Commit (#151f312) reproduces this with the remoll generator and the included .root file (needs to have its extension corrected to .root to work)

https://github.com/JeffersonLab/remoll/blob/fb0d3d98d1112c7042d90c48d37df12ef0a47b60/src/remollPrimaryGeneratorAction.cc#L110

**Comment by wdconinc** [comment]
What do you mean with 'this change'? There is the commit fb0d3d9 referenced, but that didn't affect that line in the primary generator action.

I'll take a look at your uploaded ROOT file (but please don't upload binary non-source files: they become part of the repository in perpetuity and git does not even attempt to compress them).

**Comment by cameronc137** [comment]
By 'this change' I mean the shift of the beamTarget stuff. It appears to be the only major simulation change between when I was able to get this kind of a user specified birth vertex to work and now, where there is something else muddying how I thought it worked. 

The command `evt->fVertexPos.setZ( 0.0 );` makes the beamTarget class behavior go away. When this line is not there then the vertex starting positions of new events are chosen to spread out over the range of the targetDaughter.gdml's declared beam target volume. This is fine, but I don't understand exactly why this command is needed to turn off that beam target volume business or how it works really, and I don't think I have fully gotten rid of it by just doing this one line here.

I'm checking to make sure that the code worked the way I thought it did before the beamTarget update in December.

I made sure the root file was <100kb, but yes, I agree, binaries should be avoided. Sorry.

https://github.com/JeffersonLab/remoll/blob/c45b411dd55e5145ab8dbabe6c5bbbc844161834/src/remollGenRemoll.cc#L163
